src/script/calling/CallingRepository.ts

```diff
@@ -667,20 +657,65 @@ export class CallingRepository {
             this.abortCall(conversationId, LEAVE_CALL_REASON.ABORTED_BECAUSE_FAILED_TO_UPDATE_MISSING_CLIENTS);
           }
         }
-        break;
+        return this.processCallingMessage(conversation, event);
       }
+
       case CALL_MESSAGE_TYPE.REMOTE_MUTE: {
         const call = this.findCall(conversationId);
-        if (call) {
-          this.muteCall(call, true, MuteState.REMOTE_MUTED);
+        if (!call) {
+          return;
         }
-        break;
+
+        const isSenderAdmin = conversation.isAdmin(userId);
+        if (!isSenderAdmin) {
+          return;
+        }
+
+        const selfUserId = this.selfUser?.qualifiedId;
+        const selfClientId = this.selfClientId;
+
+        if (!selfUserId || !selfClientId) {
+          return;
+        }
+
+        const isSelfClientTargetted =
+          !!content.data.targets[selfUserId.domain]?.[selfUserId.id]?.includes(selfClientId);
+
+        if (!isSelfClientTargetted) {
+          return;
+        }
+
+        this.muteCall(call, true, MuteState.REMOTE_MUTED);
+        return this.processCallingMessage(conversation, event);
```

Comment:
This is the main change - from now we verify whether a self client is included in `targets` map and whether event creator is a conversation admin.